scripts/config/msu/utils.nut

```diff
@@ -138,4 +139,35 @@
 		}
 		throw ::MSU.InvalidValue(_compareResult);
 	}
+
+	function getActiveState()
+	{
+		foreach (name, state in this.States)
+		{
+			if (state != null && state.isVisible() && name != "root_state")
+			{
+				return state;
```

Comment:
Will there always be at least one active state? If yes, then this is fine. Otherwise add handling for the case when no state is active/visible.